### PR TITLE
Firefox: Pass evaluate tests

### LINKED ISSUE #100: Pass evaluation.spec.js tests in Firefox

### PR DESCRIPTION
**Pull request by Meir017**
closes #100

### CODE REVIEW

**Comment by Meir017** [comment]
this depends on #282

**Comment by kblok** [comment]
Ready to unconflict @Meir017

**Comment by Meir017** [comment]
@kblok done

File: src/PlaywrightSharp.Firefox/FirefoxExecutionContext.cs [hunk 1]
```diff
@@ -174,28 +190,19 @@ private T DeserializeValue<T>(RemoteObject remoteObject)
...
-                RemoteObjectType.Number => value.GetDouble(),
-                RemoteObjectType.Bigint => value.GetDouble(),
-                RemoteObjectType.Boolean => value.GetBoolean(),
-                _ => value.ToObject<T>()
-            };
+                case RemoteObjectUnserializableValue.NegativeZero: return -0;
+                case RemoteObjectUnserializableValue.NaN: return double.NaN;
+                case RemoteObjectUnserializableValue.Infinity: return double.PositiveInfinity;
+                case RemoteObjectUnserializableValue.NegativeInfinity: return double.NegativeInfinity;
+                default: throw new Exception("Unsupported unserializable value: " + unserializableValue);
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/FirefoxExecutionContext.cs:204]: If we don't use expressions, let's split this in lines.

File: src/PlaywrightSharp.Firefox/FirefoxPage.cs [hunk 2]
```diff
@@ -321,6 +339,23 @@ private void OnExecutionContextDestroyed(RuntimeExecutionContextDestroyedFirefox
             }
         }
 
+        private void RemoveContextsForFrame(object sender, FrameEventArgs e)
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/FirefoxPage.cs:342]: OnPageFrameDetached?

File: src/PlaywrightSharp.Firefox/Helper/JsonHelper.cs [hunk 3]
```diff
@@ -17,7 +13,6 @@ static JsonHelper()
                 IgnoreNullValues = DefaultJsonHelper.DefaultJsonSerializerOptions.IgnoreNullValues,
                 Converters =
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/Helper/JsonHelper.cs:14]: Can we get rid of the `Converters` initializer at all?

**Comment by kblok** [comment]
@Meir017 just curious, why did you need to code all the `FrameDetached` stuff and we didn't need that for chromium?

Just code you stumbled upon?

File: src/PlaywrightSharp.Firefox/FirefoxExecutionContext.cs (hunk 1 above)
Comment (Meir017) [reply src/PlaywrightSharp.Firefox/FirefoxExecutionContext.cs:204]: for some reason when this was an expression the `-0` was returned as a `double` type and then failed to case to `int` so I changed this to be like chromium.

File: src/PlaywrightSharp.Firefox/FirefoxPage.cs (hunk 2 above)
Comment (Meir017) [reply src/PlaywrightSharp.Firefox/FirefoxPage.cs:342]: this https://github.com/kblok/playwright/blob/v0.10.0/src/firefox/ffPage.ts#L57 & https://github.com/kblok/playwright/blob/v0.10.0/src/firefox/ffPage.ts#L127

**Comment by Meir017** [comment]
@kblok that's another way of how firefox and webkit use to remove context ids from the map on the PageDelegate object. chromium is the clean implementation here ;)

**Comment by Meir017** [comment]
hmm, chromium on windows 
`PlaywrightSharp.Tests.ElementHandle.ElementHandleBoundingBoxTests.ShouldWork` is failing

Error Message:
   Assert.Equal() Failure
Expected: Rect { Height = 50, Width = 50, X = 100, Y = 50 }
Actual:   Rect { Height = 50, Width = 50, X = 100, Y = 59 }
  Stack Trace:
     at PlaywrightSharp.Tests.ElementHandle.ElementHandleBoundingBoxTests.ShouldWork() in D:\a\playwright-sharp\playwright-sharp\src\PlaywrightSharp.Tests\ElementHandle\ElementHandleBoundingBoxTestsTests.cs:line 30

**Comment by Meir017** [comment]
doesn't seem to be related to this pr

**Comment by kblok** [comment]
@Meir017 yeah I'm getting that in mine too. I wouldn't worry about that now.

**Pull request merged**